src/frontend/src/lib/utils/format.utils.ts

```diff
@@ -86,3 +86,18 @@ export const formatUSD = (
 		.format(value)
 		.replace(/,/g, '’');
 };
+
+/**
+ * Formats and if needed rounds in an Upward direction provided BigNumberish
+ *
+ * @param value BigNumberish to format.
+ * @param decimals The decimals to format with.
+ * @returns Formated and rounded number.
+ */
+export const formatBigNumberish = ({
+	value,
+	decimals
+}: {
+	value: BigNumberish;
+	decimals: number;
+}): number => Number(value) / 10 ** decimals;
```

Comment:
I would suggest to leave it in the universe of `BigNumber` and only in the final converting to `Number`, using something like `formatUnits` from `ethers` (I think we have a `parseToken` function that does something similar) and then `toFixed` if you want to round it

```
// pseudo-code
Utils.formatUnits(value, unitName:decimals).toNumber().toFixed(decimals)
```

Reply:
Just wondering what would be the benefit here? Do you think it will be more performant? It does this operation with numbers in `main`, I only extracted it in a separate function.

Reply:
Indeed, I think it was wrong from the beginning, it shouldn't have been done in `number`, it should have stayed in `BigNumberish` and then parsed

It is always safer, I think, since Typescript native `Number` type (which is a 64-bit floating-point number) cannot safely handle very large integers or perform operations on them without losing precision

Reply:
I changed the implementation to use BigNumbers instead of numbers. Tested it for small and big values and decimals, working fine. Please take a look.